cinder/volume/manager.py

```diff
@@ -642,7 +642,10 @@ def _count_host_stats(self, context, export_volumes=False):
             # for clone based snapshots.  This only counts snapshots against
             # the snapshot/source volume's pool if the snapshot is a clone
             # of the source volume.
-            if self.driver.capabilities.get('snapshot_type') == 'clone':
+            independent_snapshots = self.driver.capabilities.get(
+                'has_independent_snapshots')
+            if (self.driver.capabilities.get('snapshot_type') == 'clone' and
+                    independent_snapshots == 'true'):
```

Comment:
Why is that only the case for independent snapshots? Wouldn't clones also count in the other case, because they are always clone and thus always take space?

Reply:
clones are always accounted for because they are real volumes.   The only case we have to concern ourselves with is snapshots, because snaps from cinder's perspective don't take up space....unless it's our vmdk driver.

Reply:
Yes, sure, but if we do not have independent snapshots, then we still get snapshots as clones, right? So they still need to be accounted for.

Reply:
the independent snapshots are clones basically.  they are also allowed to be created anywhere within the shard, hence the test for both clone and independent and only the vmdk driver has those 2 to true.   fcd is false, because the snaps aren't clones.

Reply:
So the FCD driver still has independent snapshots, just not clones? Then FCD was previously already not getting that `if` to `True`.

I don't think I get what we're trying to achieve here. Any snapshot, that's a clone, takes up space _somewhere_, no matter if it can move to another host or not. So I would have assumed we need to count them - even if they're on the source volume's host.

Still, I'm not going to stand in your way. Whatever I'm missing to understand here, the code looks like it should work in general. To me it looks like it guards against counting snapshots, that are clones but not able to be on another pool.